.github/workflows/cypress.yml

```diff
@@ -0,0 +1,36 @@
+name: Cypress tests
+
+on:
+  push:
+    branches:
+      - master
+      - develop
+  pull_request:
+    branches:
+      - master
+      - develop
+
+jobs:
+  cypress-run:
+    name: Run end-to-end tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
chore (blocking): this ref will not exist if this is not a pull request but the tirggers are also on push. You will need to add a separate job for on push

Reply:
For push I suggest using the same approach they use here: https://nx.dev/ci/monorepo-ci-github-actions defining 2 steps one for push (lines: 13-24) and one for pull request (line 26) you have the PR one but we will need the other as well to ensure that nx affected also works properly on push

Reply:
Done!